lib/Doctrine/Deprecations/Deprecation.php

```diff
@@ -263,4 +269,23 @@ public static function getTriggeredDeprecations(): array
     {
         return self::$ignoredLinks;
     }
+
+    private static function getType(): int
```

Comment:
Nit: let's make clear where we're getting the type from.

```suggestion
    private static function getTypeFromEnv(): int
```

Reply:
Also, `@return self::TYPE_*` please :pray: